src/openforms/authentication/tests/test_signals.py

```diff
@@ -67,3 +73,79 @@ def test_attribute_set_for_demo_plugin_with_staff_user(self):
 
         instance.refresh_from_db()
         self.assertEqual(instance.auth_info.value, "123")
+
+    @override_settings(
+        CORS_ALLOW_ALL_ORIGINS=False, CORS_ALLOWED_ORIGINS=["http://foo.bar"]
+    )
+    def test_successful_auth_sends_signal(self):
+        register = Registry()
+        register("plugin1")(Plugin)
+        plugin = register["plugin1"]
+
+        step = FormStepFactory(
+            form__slug="myform",
+            form__authentication_backends=["plugin1"],
+            form_definition__login_required=True,
+        )
+        form = step.form
+
+        # we need an arbitrary request
+        factory = RequestFactory()
+        init_request = factory.get("/foo")
+        url = plugin.get_return_url(init_request, form)
+        next_url = "http://foo.bar"
+        request = factory.get(url, {"next": next_url})
+
+        # Add session information to the request:
+        request.session = {
+            FORM_AUTH_SESSION_KEY: {
+                "plugin": "plugin1",
+                "attribute": Plugin.provides_auth,
+                "value": "123",
+            }
+        }
+
+        return_view = AuthenticationReturnView.as_view(register=register)
+
+        with patch(
+            "openforms.authentication.views.authentication_success.send"
+        ) as m_send:
+            response = return_view(request, slug=form.slug, plugin_id=plugin.identifier)
+
+            self.assertEqual(response.status_code, 302)
+            m_send.assert_called()
```

Comment:
I'd make this a bit more specific - `m_send.assert_called_once_with(request=request)`

Reply:
The request is of a different class, so I formulated the assertion a bit differently